### ISSUE TITLE
Parser

### ISSUE THREAD

**Issue by scorsi** [Horilang/Horilang#1]
### Expected behaviour
When a node (in the building step) remove an other node from the nodes list, it should be replicated in the Parser

### Actual behaviour
Modification on the nodes list aren't replicated into the Parser

**Comment by scorsi** [comment]
Maybe instead of 
```groovy
nodes = nodes.dropRight(1)
```
we can do
```groovy
nodes.remove(node)
```

**Comment by scorsi** [comment]
Fixed in b2c07bb289b2fd5a3cb173ab307181d3af3e2fba.
Finally did:
```groovy
nodes.remove(nodes.last())
```
